specs/driver_spec.rb

```diff
@@ -58,23 +60,68 @@
-      trip = RideShare::Trip.new({id: 8, driver: @driver, passenger: nil, date: "2016-08-08", rating: 5})
+      start_time = Time.parse("2016-05-02T09:06:00+00:00")
+      end_time = Time.parse("2016-05-02T09:56:00+00:00")
+      trip = RideShare::Trip.new({id: 8, driver: @driver, passenger: nil, start_time: start_time, end_time: end_time, cost: 3, rating: nil})
+      # trip2 = RideShare::Trip.new({id: 9, driver: @driver, passenger: nil, start_time: start_time, end_time: end_time, cost: 2, rating: 5})
       @driver.add_trip(trip)
+      # @driver.add_trip(trip2)
     end
 
-    it "returns a float" do
-      @driver.average_rating.must_be_kind_of Float
+    # it "returns a float" do
+    #   @driver.average_rating.must_be_kind_of Float
+    # end
+
+    # it "returns a float within range of 1.0 to 5.0" do
+    #   average = @driver.average_rating
+    #   average.must_be :>=, 1.0
+    #   average.must_be :<=, 5.0
+    # end
+    #
+    # it "returns zero if no trips" do
+    #   driver = RideShare::Driver.new(id: 54, name: "Rogers Bartell IV", vin: "1C9EVBRM0YBC564DZ")
+    #   driver.average_rating.must_equal 0
+    # end
+
+    it "averages rating while excluding nil ratings" do
+      # Assemble
+      driver = RideShare::Driver.new(id: 54, name: "Rogers Bartell IV", vin: "1C9EVBRM0YBC564DZ")
+      start_time = Time.parse("2016-05-02T09:06:00+00:00")
+      end_time = Time.parse("2016-05-02T09:56:00+00:00")
+      trip = RideShare::Trip.new({id: 8, driver: @driver, passenger: nil, start_time: start_time, end_time: end_time, cost: 3, rating: nil})
+      trip2 = RideShare::Trip.new({id: 9, driver: @driver, passenger: nil, start_time: start_time, end_time: end_time, cost: 2, rating: 5})
+      driver.add_trip(trip)
+      driver.add_trip(trip2)
+      average = driver.average_rating
+      # Act/Assert
+      average.must_equal(5)
     end
+  end
+
+  describe "total revenue" do
+    before do
+      end_time = Time.parse("2016-05-02T09:56:00+00:00")
+      @trips = [
+        RideShare::Trip.new({end_time: end_time, cost: 5, rating: 3}),
+        RideShare::Trip.new({end_time: end_time, cost: 7, rating: 3}),
+        RideShare::Trip.new({end_time: end_time, cost: 8, rating: 3}),
+      ]
+      driver_data = {
+        id: 7,
+        vin: '12345678912345678',
+        name: 'test driver',
+        trips: @trips
+      }
+      @driver = RideShare::Driver.new(driver_data)
 
-    it "returns a float within range of 1.0 to 5.0" do
-      average = @driver.average_rating
-      average.must_be :>=, 1.0
-      average.must_be :<=, 5.0
     end
 
-    it "returns zero if no trips" do
-      driver = RideShare::Driver.new(id: 54, name: "Rogers Bartell IV", vin: "1C9EVBRM0YBC564DZ")
-      driver.average_rating.must_equal 0
+    it "returns the total revenue for a trip" do
+      @driver.total_revenue.must_equal(12.04)
+    end
+
+    it "returns the average revenue per hour for a trip" do
+      @driver.avg_revenue.must_equal(0.07)
     end
```

Comment:
You should break tests for different methods out into different `describe` blocks.